notary-commons/src/main/kotlin/sidechain/iroha/consumer/IrohaConsumerImpl.kt

```diff
@@ -72,50 +81,68 @@ class IrohaConsumerImpl(
      * @return list of hex hashes that were accepted by iroha
      */
     override fun send(lst: Iterable<TransactionOuterClass.Transaction>): Result<List<String>, Exception> {
-        irohaAPI.transactionListSync(lst)
         return Result.of {
-            lst.map {
-                Utils.hash(it)
-            }
-        }.map { hashes ->
-            hashes.map {
-                checkTransactionStatus(it)
+            val successfulTxHashesList = ArrayList<String>()
+            irohaAPI.transactionListSync(lst)
+            lst.map { tx -> Utils.hash(tx) }.forEach { txHash ->
+                val txStatus = TxStatus.createEmpty()
+                val observer = createTxStatusObserver(txStatus)
+                irohaAPI.txStatus(txHash).blockingSubscribe(observer)
```

Comment:
Can a final status receiving be guaranteed ?

Reply:
No. For example, we may wait for not existing tx hash